app/submitter/converter.py

```diff
@@ -89,7 +89,10 @@ def convert_answers_to_data(answer_store, questionnaire_json, routing_path):
         answer_schema_list = SchemaHelper.get_answers_by_id_for_block(block_json)
 
         for answer in answers_in_block:
-            answer_schema = answer_schema_list[answer['answer_id']]
+            try:
+                answer_schema = answer_schema_list[answer['answer_id']]
+            except KeyError:
+                answer_schema = None
```

Comment:
Can we get some error logging here

Reply:
ideally with some info about the actual id/answer causing the issue

Reply:
I think we should add logging as @collisdigital mentions. Also given that the next statement is `if answer_schema is not None` we should just `return data` and not bother evaluating the rest of the code?